rftp/server2.go

```diff
@@ -34,23 +34,35 @@ type clientConnection struct {
 func (c *clientConnection) writeResponse() {
 	log.Println("start writing response packets")
 	lastAck := uint8(0)
+	rateControl := &aimd{congRate: 4}
+	rateControl.start()
+	defer rateControl.stop()
 	for {
+		// TODO: Wait for rateControl.available() to be true. How should the caller
+		// of rateControl be informed? ACKs should be handled even if
+		// !rateControl.available().
```

Comment:
Maybe something like a separate go routine doing passive waiting on `rateControl.available()` and writing to some channel (lets call it `availableChan`) when it returns true. Then something like this could block the loop until we can continue:
```
availableChan := make(chan struct{})
go func() {
	while !rateControl.available() {
		time.Sleep(1 * time.Secnod) // This is probably not the best way to do this
	}
	availableChan <- struct{}{}
}()
select {
	case <- availableChan:
	case ack := <-c.ack:
		lastAck = ack.ackNumber
		rateControl.onACK(ack)
 		c.reschedule(ack)
		continue
}
```
The `availableChan` will make the select fall through to the second select, while an ACK will just be handled and restart the loop, to wait for more ACKs or budget.
Maybe it's also helpful, if available returns a channel, on which we can select directly. This channel could be internal to `rateControl` and written to every tick and everytime an ACK raises the cong and flow rates above the currently used budget.